### PR TITLE
Add merge accounts logic

### PR DESCRIPTION
**Pull request by aaronc**

### CODE REVIEW

File: merge_test.go [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+			nil,
+			true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := MergeAccounts(tt.accounts)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("MergeAccounts() error = %v, wantErr %v", err, tt.wantErr)
+				return
```
Comment (clevinson) [inline merge_test.go:357]: This return was not actually exiting, so i added a `!tt.wantErr` check below.

File: merge.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		existing, ok := distMap[t]
+		var newAmount apd.Decimal
+		if ok {
+			_, err := apd.BaseContext.Add(&newAmount, &existing.Regen, &amount)
+			if err != nil {
+				return Account{}, err
+			}
+		} else {
+			newAmount = amount
+		}
```
Comment (clevinson) [inline merge.go:70]: This else block was missing. I added after catching in the test.

File: merge.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		var newAcc Account
+		if ok {
+			var err error
+			newAcc, err = mergeTwoAccounts(acc, existing)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			newAcc = acc
+		}
```
Comment (clevinson) [inline merge.go:29]: This else block was missing, i added after catching in the test.

**Review by clevinson — COMMENTED** [review]
Tests added, I changed a few pieces of logic to fix failing tests. Can you review / ack @aaronc ?

**Review by aaronc — COMMENTED** [review]
ACK

**Review by clevinson — APPROVED** [review]
ack

**Pull request merged**